pages/composer.js

```diff
@@ -1,97 +1,132 @@
-import ExportModal from "../src/components/HikeComposer/ExportModal";
-import EditHike from "../src/components/HikeComposer/EditHike";
-import { useRouter } from "next/router";
-import PreviewSwitcher from "../src/components/HikeComposer/PreviewSwitcher";
 
 const { TextArea } = Input;
 const { Option } = Select;
 
-const BASE_API = "http://localhost:3200/api";
+const BASE_API = 'http://localhost:3200/api';
 
-const GenerateTourPage = () => {
-  const { query } = useRouter();
-  const { checksum, kuid, zipName } = query;
-  const [values, setValues] = useState({
-    hikeVersion: "1.0",
-    title: "",
-    description: "",
+const localizedFields = ['title', 'description', 'details'];
+const getDefaultLocalizationObj = () => locales.reduce((previous, locale) => ({
+  ...previous,
+  [locale]: {},
+}), {});
+const getNewTourTemplate = ({ checksum, kuid, zipName }) => {
+  const timestamp = Math.floor(Date.now() / 1000);
+  return {
+    nid: timestamp,
+    created: timestamp,
+    updated: timestamp,
+    hikeVersion: '1.0',
+    title: '',
+    description: '',
     checksum,
     kuid,
     category: [],
-    details: "",
+    details: '',
     cards: 1,
     tourLink: zipName,
     time: 1,
     platformVersion: 9.2,
-  });
+    localization: getDefaultLocalizationObj(),
+  };
+};
+
+const GenerateTourPage = () => {
+  const { query } = useRouter();
+  const { checksum, kuid, zipName } = query;
+  const [values, setValues] = useState(getNewTourTemplate({ checksum, kuid, zipName }));
 
   const [isCKEditorVisible, setIsCKEditorVisible] = useState(false);
-  const [view, setView] = useState("card");
-  const [selectedFile, setSelectedFile] = useState({});
+  const [view, setView] = useState('card');
+  const [selectedFile/* , setSelectedFile */] = useState({});
   const [selectedCategory, setSelectedCategory] = useState({});
-  const [previewMode, setPreviewMode] = useState("split");
+  const [previewMode, setPreviewMode] = useState('split');
   const [visible, setVisible] = useState(false);
   const [exportJsonData, setexportJsonData] = useState({});
 
   const [categories, setCategories] = useState([]);
 
-  //Form Switcher states
-  const [current, setCurrent] = useState("new");
+  // Form Switcher states
+  const [current, setCurrent] = useState('new');
+
+  // Editing Language switch
+  const [editingLanguage, setEditingLanguage] = useState(defaultLocale);
 
   const handleSwitcherClick = (e) => {
     setCurrent(e.key);
   };
 
   const onChangeCategory = (categoryName) => {
     const selected = categories.find(
-      (category) => categoryName === category.categoryName
+      category => categoryName === category.categoryName,
     );
     setSelectedCategory(selected);
   };
 
-  const changePreview = (view) => {
-    setView(view);
-    if (view === "list") {
-      setPreviewMode("split");
+  const changePreview = (v) => {
+    setView(v);
+    if (visible === 'list') {
```

Comment:
I think this should be

if (v === 'list')

Reply:
fixed